src/liberty/libertyProject.ts

```diff
@@ -203,6 +203,56 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 		statusMessage.dispose();
 	}
 
+	/**
+	 * This method asks the user to save the workspace first if it is untitled and the workspace contains more than
+	 * one project. If the workspace is not saved, there are chances that the project's state may not be saved and 
+	 * manually added projects may not be visible in the Liberty dashboard in a new VS Code session.
+	 */
+	public async checkUntitledWorkspaceAndSaveIt(): Promise<void> {
+		return new Promise((resolve) => {
+			try {
+				vscode.window.showInformationMessage(
+					localize("workspace.not.saved.projects.may.not.persist"),
+					{ modal: true },
+					'Save Workspace'
+				).then(async (selection) => {
+					if (selection === 'Save Workspace') {
+						/**
+						 * setting workspaceSaveInProgress to true and storing it in globalstate for identifyting that the
+						 * workspace is saved and needs to save the manually added projects to the dashboard
+						 */
+						await this._context.globalState.update('workspaceSaveInProgress', true);
+						//opens the saveWorkspace as dialog box
+						await vscode.commands.executeCommand('workbench.action.saveWorkspaceAs');
+					}
+					/**
+					 * If the user cancels saving the workspace and exits without saving, the data stays in the global state, 
+					 * which is shared across all VS Code instances. To prevent this data from being mistakenly used in other 
+					 * sessions and added to the dashboard, it should be cleared if the user cancels the save.
+					 */
+					util.clearDataSavedInGlobalState(this._context);
+					resolve();
+				});
+			} catch (error) {
+				util.clearDataSavedInGlobalState(this._context);
+				resolve();
+				console.debug("exception while saving the workspace" + error);
```

Comment:
In scenarios like this, it is typically good practice to place the debug statement first before calling other methods just in case those methods run into other problems. That way you at least know you entered this code block.

Reply:
rearranged the code